pkg/util/prefix_wildcard.go

```diff
@@ -0,0 +1,15 @@
+package util
+
+import "strings"
+
+// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\*|-\*)?$`
+
+// PrefixWildcard is a string that supports globbing at its end.
+type PrefixWildcard string
+
+// Matches returns true if the candidate parameter is either an exact match of the PrefixWildcard,
+// or if the PrefixWildcard is a valid glob-match for the candidate.  The PrefixWildcard must
+// end in a "*" to be considered a glob.
+func (pw PrefixWildcard) Matches(candidate string) bool {
+	return strings.HasPrefix(candidate, strings.TrimSuffix(string(pw), "*"))
```

Comment:
We need to perform a test for an exact match if there is no `*` suffix. We should only perform prefix matching if the user specifies a glob.

Reply:
Fixed this case.